### DISCUSSION TITLE
Support for SRIOV CNI

### DISCUSSION THREAD

**Discussion by sivakumars28** [containerd/nerdctl#2846] [General]
I have tried to add the sriov plugin in the cni location (/opt/cni/bin), and tried to create a network, but it is allowing only to create network with standard plugins. In case if we need to use our custom CNI, how it can be supported from nerdctl?

**Comment by AkihiroSuda** [comment]
You can modify `/etc/cni/net.d/nerdctl-*.conflist` via any text editor

**Reply by sivakumars28** [reply]
I can see the default one here 

{
  "cniVersion": "1.0.0",
  "name": "bridge",
  **"nerdctlID": "17f29b073143d8cd97b5bbe492bdeffec1c5fee55cc1fe2112c8b9335f8b6121",**
  "nerdctlLabels": {},
  "plugins": [
    {
      "type": "bridge",
      "bridge": "nerdctl0",
      "isGateway": true,
      "ipMasq": true,
      "hairpinMode": true,
      "ipam": {
        "ranges": [
          [
            {
              "gateway": "10.4.0.1",
              "subnet": "10.4.0.0/24"
            }
          ]
        ],
        "routes": [
          {
            "dst": "0.0.0.0/0"
          }
        ],
        "type": "host-local"
      }
    },
    {
      "type": "portmap",
      "capabilities": {
        "portMappings": true
      }
    },
    {
      "type": "firewall",
      "ingressPolicy": "same-bridge"
    },
    {
      "type": "tuning"
    }
  ]
}


How we can define "nerdctlID": "17f29b073143d8cd97b5bbe492bdeffec1c5fee55cc1fe2112c8b9335f8b6121", for my new plugin, As I am trying to do SRIOV https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin

**Reply by AkihiroSuda** [reply]
ID is computed like this, but you can (probably) leave it unspecified
https://github.com/containerd/nerdctl/blob/0a20db6396185de885afa5af3df9f01ceb3e12a9/pkg/netutil/netutil.go#L464-L467

**Reply by sivakumars28** [reply]
Thanks, I tried but ending up in same error 

error while executing [/usr/local/bin/nerdctl network create --label=com.docker.compose.project=nginx --label=com.docker.compose.network=node-network --driver=sriov nginx_node-network]: "time=\"2024-03-01T17:19:40Z\" level=warning msg=\"duplicate network name \\\"bridge\\\", &netutil.NetworkConfig{NetworkConfigList:(*libcni.NetworkConfigList)(0xc00017f6e0), NerdctlID:(*string)(0xc0004679a0), NerdctlLabels:(*map[string]string)(0xc000168be8), File:\\\"/etc/cni/net.d/nerdctl-bridge.conflist\\\"} will get superseded by &netutil.NetworkConfig{NetworkConfigList:(*libcni.NetworkConfigList)(0xc00017f920), NerdctlID:(*string)(0xc000960530), NerdctlLabels:(*map[string]string)(0xc000168c48), File:\\\"/etc/cni/net.d/nerdctl-sriov.conflist\\\"}\"\ntime=\"2024-03-01T17:19:40Z\" level=fatal msg=\"unsupported cni driver \\\"sriov\\\"\"\n": exit status 1


As I can see in code here https://github.com/containerd/nerdctl/blob/main/pkg/netutil/netutil_unix.go , where it has only two cases it supports

case "bridge":
case "macvlan", "ipvlan":
rest all it goes to return nil, fmt.Errorf("unsupported cni driver %q", driver)

If you help me to understand how we can put more plugin and make it work, It will be helpful to add sriov.
